File: 118-Pascal's_Triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    if (!numRows) return [];

    let output = [[1]];

    for (let i = 1; i < numRows; i++) {
        let row = [1];
        for (let j = 1; j < i; j++) {
            row.push(output[i - 1][j - 1] + output[i - 1][j]);
        }
        row.push(1);
        output.push(row);
    }

    return output;
};

File: 118-Pascal's_Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []
        for i in range(numRows):
            numRow = [1] * (i + 1)
            for j in range(1, i):
                numRow[j] = res[i - 1][j - 1] + res[i - 1][j]
            res.append(numRow)
        return res
